app/Models/GenereRepository.php

```
@@ -171,10 +173,34 @@ public function delete(int $id): bool
+        // Physical shelf bindings must block a plain delete rather than silently
+        // cascade. posizioni.genere_id is ON DELETE CASCADE: deleting the genre
+        // would drop its posizioni and NULL libri.posizione_id even for books of
+        // OTHER genres that happen to be shelved there (they are not caught by the
+        // genere_id/sottogenere_id book check above). mensole.genere_id is RESTRICT,
+        // so an assigned mensola would otherwise crash with FK error 1451.
+        $stmt = $this->db->prepare("SELECT COUNT(*) as cnt FROM posizioni WHERE genere_id = ?");
+        $stmt->bind_param('i', $id);
+        $stmt->execute();
+        $count = (int)$stmt->get_result()->fetch_assoc()['cnt'];
+        if ($count > 0) {
+            throw new \RuntimeException('Impossibile eliminare: il genere ha posizioni fisiche assegnate');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '187', column '23'). (undefined)

(MissingImport)